admob/src/android/admob_android.cc

```diff
@@ -441,33 +453,151 @@ JNIEnv* GetJNI() {
+    adr->is_successful = false;
+  } else if (adr->code != kAdMobErrorNone) {
+    // C++ SDK Android AdMob Wrapper encountered an error.
+    adr->is_wrapper_error = true;
+    adr->is_successful = false;
+    adr->message = error_message;
+    adr->domain = "SDK";
+    adr->to_string = std::string("Internal error: ") + adr->message;
+    future_error_message = adr->message;
+  }
+
+  // Invoke a friend of LoadAdResult to have it invoke the LoadAdResult
+  // protected constructor with the LoadAdResultInternal data.
+  AdmobInternal::CompleteLoadAdFuture(
+      callback_data, adr->code, future_error_message, load_ad_result_internal);
+}
+
+void CompleteLoadAdAndroidErrorResult(JNIEnv* env, jlong data_ptr,
+                                      jobject j_load_ad_error,
+                                      AdMobError error_code,
+                                      jstring j_error_message) {
+  FIREBASE_ASSERT(env);
+  FIREBASE_ASSERT(data_ptr);
+  FIREBASE_ASSERT(j_error_message);
+
+  FutureCallbackData<LoadAdResult>* callback_data =
+      reinterpret_cast<firebase::admob::FutureCallbackData<LoadAdResult>*>(
+          data_ptr);
+
+  std::string error_message = util::JStringToString(env, j_error_message);
+
+  CompleteLoadAdCallback(callback_data, j_load_ad_error, error_code,
+                         error_message);
+}
+
+void CompleteLoadAdInternalResult(
+    FutureCallbackData<LoadAdResult>* callback_data, AdMobError error_code,
+    const char* error_message) {
+  FIREBASE_ASSERT(callback_data);
+  FIREBASE_ASSERT(error_message);
+
+  CompleteLoadAdCallback(callback_data, /*j_load_ad_error=*/nullptr, error_code,
+                         error_message);
+}
+
+extern "C" JNIEXPORT void JNICALL
+Java_com_google_firebase_admob_internal_cpp_BannerViewHelper_completeBannerViewLoadedAd(  // NOLINT
+    JNIEnv* env, jclass clazz, jlong data_ptr) {
+  FIREBASE_ASSERT(env);
+  FIREBASE_ASSERT(data_ptr);
+
+  FutureCallbackData<LoadAdResult>* callback_data =
+      reinterpret_cast<firebase::admob::FutureCallbackData<LoadAdResult>*>(
+          data_ptr);
+
+  CompleteLoadAdInternalResult(callback_data, kAdMobErrorNone,
+                               /*error_message=*/"");
+}
+
+extern "C" JNIEXPORT void JNICALL
+Java_com_google_firebase_admob_internal_cpp_BannerViewHelper_completeBannerViewLoadAdError(  // NOLINT
+    JNIEnv* env, jclass clazz, jlong data_ptr, jobject j_load_ad_error,
+    jint j_error_code, jstring j_error_message) {
+  FIREBASE_ASSERT(env);
+  FIREBASE_ASSERT(data_ptr);
+  FIREBASE_ASSERT(j_error_message);
+
+  const AdMobError error_code =
+      MapAndroidAdRequestErrorCodeToCPPErrorCode(j_error_code);
+
+  CompleteLoadAdAndroidErrorResult(env, data_ptr, j_load_ad_error, error_code,
+                                   j_error_message);
+}
+
+// Internal Errors use AdMobError codes.
 extern "C" JNIEXPORT void JNICALL
-Java_com_google_firebase_admob_internal_cpp_BannerViewHelper_completeBannerViewFutureCallback(
```

Comment:
FYI this is an opportunity to rename these native methods. Since we have to call RegisterNatives on them anyway (due to the way we load the Android classes, they aren't bound automatically), we could shorten / simplify the names for readability. Up to you.

Reply:
I will do it in a followup PR since I have another change based on this branch and I'd like to prevent as many merge conflicts as possible.